### PR TITLE
Do additional consolidation with mixins and remove unused CSS

### PR DESCRIPTION
**Pull request by lyzadanger**
Part of #2245 

This PR is a slightly grab-bag set of cleanup commits that improve usage of mixins, consolidate some patterns and remove some unneeded or unused CSS.

* Removed `annotation-action-bar` component style module because its only style could be replaced with a utility class
* Removed some classes that are not being used at all
* Make `FocusedModeHeader` use `card-frame` mixin instead of re-defining a card-like pattern
* Refactor `HelpPanel`'s styling — the reason this got a dive here was because I was using it as an exemplar in last week's discussion
* Remove the `form-input` component style module and move those rules into the `forms` mixins — this reflects our move toward having reusable mixins but not component style modules for things that aren't actually components

Following PRs won't be as here-and-there. Thanks for your patience!

### CODE REVIEW

File: src/styles/mixins/buttons.scss [hunk 1]
```diff
@@ -133,3 +133,15 @@
     text-decoration: underline;
   }
 }
+
```
Comment (LMS007) [inline src/styles/mixins/buttons.scss:136]: I think a comment here could be helpful (like the button--link mixin above) so its clear where it should be used.

File: src/styles/sidebar/components/focused-mode-header.scss [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
 
 // A dark grey button used for the primary action
 // in a form
 .focused-mode-header {
+  @include molecules.card-frame;
   @include layout.row($align: center);
-  @include utils.border;
   position: relative;
 
   background-color: var.$color-background;
```
Comment (LMS007) [inline src/styles/sidebar/components/focused-mode-header.scss:15]: background-color is already set with  @include molecules.card-frame;

File: src/styles/util.scss [hunk 3]
```diff
@@ -10,6 +10,15 @@
   @include layout.row;
 }
 
+.u-layout-row--align-center,
```
Comment (LMS007) [inline src/styles/util.scss:13]: Are these going to be used later somewhere?

Comment (LMS007) [inline src/styles/util.scss:13]: Are these `u-*` rules going to be used later somewhere?

File: src/styles/mixins/buttons.scss (hunk 1 above)
Comment (lyzadanger) [reply src/styles/mixins/buttons.scss:136]: Great feedback!

File: src/styles/sidebar/components/focused-mode-header.scss (hunk 2 above)
Comment (lyzadanger) [reply src/styles/sidebar/components/focused-mode-header.scss:15]: Nice catch!

File: src/styles/util.scss (hunk 3 above)
Comment (lyzadanger) [reply src/styles/util.scss:13]: If they're not used in this set of changes, they may have gotten split up from their use when I reordered some commits. They're used somewhere!

After I land the queue of changes I've got, I'll grep to make sure all of the utility classes are used. The `u-*` prefix helps with that!

**Pull request merged**